### PR TITLE
Feature doris on es

### PR DESCRIPTION
**Pull request by wuyunfeng**

### CODE REVIEW

File: be/src/util/es_scan_reader.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+public:
+    static constexpr const char* KEY_USER_NAME = "user";
+    static constexpr const char* KEY_PASS_WORD = "passwd";
+    static constexpr const char* KEY_INDEX = "index";
+    static constexpr const char* KEY_TYPE = "type";
+    static constexpr const char* KEY_SHARDS = "shards";
+    static constexpr const char* KEY_QUERY = "query";
+    static constexpr const char* KEY_BATCH_SIZE = "batch_size";
+    ESScanReader(const std::string& target, uint16_t size, std::map<std::string, std::string>& props);
```
Comment (imay) [inline be/src/util/es_scan_reader.h]:
```suggestion
    ESScanReader(const std::string& target, uint16_t size, const std::map<std::string, std::string>& props);
```

File: be/src/util/es_search_builder.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    
+}
+
+std::string SearchRequestBuilder::build() {
+    rapidjson::Document es_query_dsl;
+    rapidjson::Document::AllocatorType &allocator = es_query_dsl.GetAllocator();
+    es_query_dsl.SetObject();
+    if (_fields.size() > 0) {
+        rapidjson::Value source_node(rapidjson::kArrayType);
+        for (auto iter = _fields.cbegin(); iter != _fields.cend(); iter++) {
```
Comment (imay) [inline be/src/util/es_search_builder.cpp]: why do you call `cbegin` other `begin`. Could you add some comment for this?

File: be/src/util/es_search_builder.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+#pragma once
+#include<string>
+#include<vector>
+
+namespace doris {
+
+class SearchRequestBuilder {
+
+private:
```
Comment (imay) [inline be/src/util/es_search_builder.h]: We prefer `public` ahead of `private`

File: be/src/util/es_scan_reader.cpp [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    _network_client.set_content_type("application/json");
+    // phase open, we cached the first response for `get_next` phase
+    _network_client.execute_post_request(_query, _cached_response);
+    long status = _network_client.get_http_status();
+    if (status != 200) {
+        LOG(WARNING) << "invalid response http status for open: " << status;
+        return Status(*_cached_response);
+    }
+    VLOG(1) << "open _cached response: " << *_cached_response;
+    rapidjson::Document document_node;
```
Comment (imay) [inline be/src/util/es_scan_reader.cpp]: You'd better to write a EsResultParser to parse json, and it is easy to do unit test for this parser

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/util/es_scan_reader.cpp (hunk 4 above)
Comment (wuyunfeng) [reply be/src/util/es_scan_reader.cpp]: ESScanReader only obtain scroll id and hits document size, not responsible for parse json

File: be/src/util/es_search_builder.cpp (hunk 2 above)
Comment (wuyunfeng) [reply be/src/util/es_search_builder.cpp]: I will replace cxxx with xxx

**Pull request closed without merging**